pkgs/development/tools/java/jprofiler/test.nix

```diff
@@ -0,0 +1,8 @@
+{ runCommand, jprofiler }:
+
+runCommand "jprofiler-test-run" {
+  nativeBuildInputs = [ jprofiler ];
+} ''
+  diff -U3 --color=auto <((jpcontroller -n -f /dev/null 2>&1 || true) | head -n 1 ) <(echo 'No profiled JVMs found.')
+  touch $out
```

Comment:
This way of running `jpcontroller` forces it to load platform-related libraries via JNI and confirm that it is relinked on the right way.

To be honest this test allows to discover that:
 - on Linux is should be relinked against X11;
 - on macOS it should be done with `dontStrip = true` otherwise a signature won't pass and it fails.

The second one is tricky because UI is started and it seems working, but the real attempt to use it leads to fail.